### DISCUSSION TITLE
This proyect but raspberry pi or arduino uno version

### DISCUSSION THREAD

**Discussion by KufitaX** [sanni/cartreader#279] [General Chat]
Is there a raspberry pi or arduino uno version of your project?

**Comment by sanni** [comment]
Actually this project was started on an Arduino Uno but quickly ran out of pins and space there and was moved to the Mega.
You can still find the original Arduino Uno stuff here: https://forum.arduino.cc/t/rom-reader-for-super-nintendo-super-famicom-game-cartridges/155260

For Raspberry Pi there seems to be a GBA reader available: https://github.com/cubicstyle/raspiadvrw

**Reply by Dakkaron** [reply]
A Raspberry Pi without anything in between will also have trouble with the pin count...
@KufitaX Why would you want a Raspberry Pi/Uno version of this?

What could be done would be to attach this to a Pi and stream the data over serial, instead of to/from the SD card.

**Reply by KufitaX** [reply]
For i have RPI and Uno, and i think in rpi its more easy.

**Reply by KufitaX** [reply]
> Actually this project was started on an Arduino Uno but quickly ran out of pins and space there and was moved to the Mega. You can still find the original Arduino Uno stuff here: https://forum.arduino.cc/t/rom-reader-for-super-nintendo-super-famicom-game-cartridges/155260
> 
> For Raspberry Pi there seems to be a GBA reader available: https://github.com/cubicstyle/raspiadvrw

I found this proyect yesterday, i dont know its yours!!!

**Reply by Dakkaron** [reply]
As sanni said: Have you counted how many pins a Raspberry Pi has? Now count, how many this project needs.

If you use a Raspberry Pi you don't need to reserve any pins for display, SD card or input, since you can just use HDMI, USB and the built-in SD card. That means, you can use all 24 pins that the Raspberry Pi GPIO offers (the other pins are Ground, 5V, 3V and other fixed-function pins that can't be used).

The Gameboy needs at least 27 pins (if I count correctly).
The other platforms need more pins.

So, unless you can add a few more pins to the Raspberry Pi, or build a PCB with a fancy port expander setup, you'd probably be best off to pony up the $12 that it takes to get a Arduino Mega.

If you don't want to spend much, have a look on Aliexpress or Ebay. Sometimes you can also find a cheap one on Amazon.

All in all, the time and effort (and additional hardware) it would take to convert that to one of the devices that you own, would be probably much more than what you are going to save. Just roughly estimating: If you would be paying for the work that would take, you'd be looking at around €1000 or more.

This is already the cheap option, just go and get it.
